modules/scylladb/scylladb.go

```diff
@@ -0,0 +1,159 @@
+package scylladb
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	port           = "9042/tcp"
+	shardAwarePort = "19042/tcp"
+)
+
+// Container represents a ScyllaDB container type used in the module
+type Container struct {
+	testcontainers.Container
+}
+
+// WithConfig sets the YAML config file as an [io.Reader] to be used for the ScyllaDB container
+func WithConfig(r io.Reader) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			Reader:            r,
+			ContainerFilePath: "/etc/scylla/scylla.yaml",
+			FileMode:          0o644,
+		}
+		req.Files = append(req.Files, cf)
+
+		return nil
+	}
+}
+
+// WithShardAwareness enable shard-awareness in the ScyllaDB container so you can use the `19042` port.
+func WithShardAwareness() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.ExposedPorts = append(req.ExposedPorts, shardAwarePort)
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(shardAwarePort))
+		return nil
+	}
+}
+
+// WithAlternator enables the Alternator (DynamoDB Compatible API) service in the ScyllaDB container.
+// It will set the "alternator-port" parameter to the specified port.
+// It will also set the "alternator-write-isolation" parameter to "always" as a command line argument to the container.
+func WithAlternator(alternatorPort uint16) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		alternatorPortStr := strconv.FormatInt(int64(alternatorPort), 10)
+		req.ExposedPorts = append(req.ExposedPorts, alternatorPortStr)
+		req.WaitingFor = wait.ForAll(req.WaitingFor, wait.ForListeningPort(nat.Port(alternatorPortStr)))
+		setCommandFlag(req, map[string]string{
+			"--alternator-port":            alternatorPortStr,
+			"--alternator-write-isolation": "always",
+		})
+
+		return nil
+	}
+}
+
+// WithCustomCommands sets custom commands with  values for the ScyllaDB container.
+// This is an option to overwrite the default commands with a custom one.
+// See more [here](https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/tips/best-practices-scylla-on-docker.html)
+func WithCustomCommands(cmds map[string]string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		setCommandFlag(req, cmds)
+		return nil
+	}
+}
+
+// ConnectionHost returns the host and port of the Scylladb container with the given port
+// and obtaining the host and exposed port from the container
+// Eg: "host:port" -> "localhost:9042" -> "localhost:19042" -> "localhost:8000"
+func (c Container) ConnectionHost(ctx context.Context, port uint16) (string, error) {
```

Comment:
question: as the ports can vary would be better to provide the distinct methods for each type of port so the user doesn't have to keep track of them?

Reply:
I like the idea of simplifying the usage of both ports. I'm adding both methods